src/http/request.cr

```diff
@@ -35,7 +36,8 @@ class HTTP::Request
 
     method, path, http_version = request_line.split
     HTTP.parse_headers_and_body(io) do |headers, body|
-      return new method, path, headers, body.try &.read, http_version
+      cookies = Cookies.from_headers(headers)
```

Comment:
+1 for moving this line into `#cookies` method, that will do it only once and memoize the result. So that cookies will be parsed only when they are required - lazily.